refactor: cleanup project, remove unused dependencies, and switch to requests

mslyvynska opened the pull request:
- Removed unused dependencies (`python-dotenv`, `httpx`) from `pyproject.toml`
- Deleted unused files and dead code to streamline the codebase
- Switched to `requests` as the HTTP client library, replacing `httpx`
- Added `poetry` as the dependency manager for better package management
- Fixed issues caused by `httpx` handling responses differently (301 redirects vs. 200 status)

mslyvynska committed:
chore: remove irrelevant files, add pyproject.toml, update README
- remove unnecessary files that are not relevant to the library
-added pyproject.toml for package configuration and dependencies
- updates README.md

davidkaufmann commented on pyproject.toml:
```diff
@@ -0,0 +1,19 @@
+
+[tool.poetry.dependencies]
+python = "^3.8"
+python-dotenv = "0.19.1"
```
```suggestion
```
We're not using this

davidkaufmann commented on pyproject.toml:
```diff
@@ -0,0 +1,19 @@
+[tool.poetry.dependencies]
+python = "^3.8"
+python-dotenv = "0.19.1"
+httpx = "0.22.0"
```
```suggestion
```
We've replaced this for the requests library

davidkaufmann commented on pyproject.toml:
```diff
@@ -0,0 +1,19 @@
+python = "^3.8"
+python-dotenv = "0.19.1"
+httpx = "0.22.0"
+pydantic = "1.8.2"
```
```suggestion
```
Not using this

davidkaufmann reviewed (changes requested):
This is decent work.
Unfortunately you forgot about the constraint to filter out th estuff that we don't need.
I read the parent issue again, it wasn't clear that this should have been mandatory, sorry about that.

> use [git filter-repo](https://github.com/newren/git-filter-repo) or something similar to keep the git history

Unfortunately it doesn't explain we want to filter all other commits, files, etc and don't have them in our history.

By opening this PR, there always will be history about this.
I can't think of anything that would mitigate this.

Not sure how to handle this, let's chat tomorrow.

Edit:
Now looking at my branch and viewing the commits, there's literally no real benefit, because other commits have references on old files we actually tried to filter.
I guess th ebest solution is to use your branch and just add a few i did within the last commit (maybe cherry pick?).
Let's tackle this tomorrow.

---

I actually did use `git filter-repo` myself, I've uploaded it to branch `filtered` for example.

mslyvynska committed:
refactor: cleanup project, remove unused dependencies, and switch to requests

- Removed unused dependencies (`python-dotenv`, `httpx`) from `pyproject.toml`
- Deleted unused files and dead code to streamline the codebase
- Switched to `requests` as the HTTP client library, replacing `httpx`
- Added `poetry` as the dependency manager for better package management
- Fixed issues caused by `httpx` handling responses differently (301 redirects vs. 200 status)

mslyvynska committed:
refactor: restore common/__init.py after accidental deletion

mslyvynska committed:
refactor: delete license

davidkaufmann reviewed (approved):
Looking good, lets get this in 👍🏻
